infra/discovery/src/apollo/resolvers.js

```diff
@@ -10,27 +12,46 @@ const web3 = new Web3(process.env.PROVIDER_URL || 'http://localhost:8545')
 const resolvers = {
   JSON: GraphQLJSON,
   Query: {
+    /**
+     * Search for listings by either terms or by ids.
+     * Non-visible listings are filtered out.
+     */
     async listings(root, args) {
-      // Get listing Ids from Elastic.
-      const { listings, stats } = await search.Listing.search(
-        args.searchQuery,
-        args.sort,
-        args.order,
-        args.filters,
-        args.page.numberOfItems,
-        args.page.offset
-      )
-      logger.info(
-        `Query: "${args.searchQuery}" returned ${listings.length} results.`
-      )
-      return {
-        nodes: listings,
-        offset: args.page.offset,
-        numberOfItems: listings.length,
-        totalNumberOfItems: stats.totalNumberOfListings,
-        stats: {
-          maxPrice: stats.maxPrice,
-          minPrice: stats.minPrice
+      if (args.ids) {
```

Comment:
Note: this code path is not used in this PR. I started adding it and then decided it was better to add a different query (visibleListingIds). But since it could be handy to query listings from discovery by ids, I'm leaving it in here for future use...